pkg/hubble/relay/observer/server.go

```diff
@@ -81,7 +82,11 @@ func (s *Server) GetFlows(req *observerpb.GetFlowsRequest, stream observerpb.Obs
 	ctx, cancel := context.WithCancel(stream.Context())
 	defer cancel()
 
-	peers := s.peers.List()
+	peers, err := filterNodes(req, s.peers.List())
```

Comment:
nit: as peers can be empty here, we can return fast here if you want

Reply:
I want to leave this as-is for the moment, as the list of peers will change dynamically in the future (#12833).